InvenTree/order/models.py

```diff
@@ -254,7 +254,11 @@ def add_line_item(self, supplier_part, quantity, group=True, reference=''):
             if matches.count() > 0:
                 line = matches.first()
 
-                line.quantity += quantity
+                # update quantity and price
+                quantity_new = line.quantity + quantity
+                line.quantity = quantity_new
+                if line.purchase_price:
+                    line.purchase_price = supplier_part.get_price(quantity_new) / quantity_new
```

Comment:
Is there any case for which `quantity_new` could be equal to 0? (eg. both `line.quantity` and `quantity` both equal to 0 or cancelling each other)

Reply:
Also maybe handle the case where `get_price` returns `None`?

Reply:
> 
> 
> Is there any case for which `quantity_new` could be equal to 0? (eg. both `line.quantity` and `quantity` both equal to 0 or cancelling each other)

No, that section is only called if there is already a line for that supplier part - which must be 0 or higher and you can not add a part with a quantity 0 or smaller (see line 241).

> 
> 
> Also maybe handle the case where `get_price` returns `None`?

Is now included from line 261 following